pkg/controller/datavolume/pvc-clone-controller.go

```diff
@@ -450,6 +450,16 @@ func (r *PvcCloneReconciler) selectCloneStrategy(datavolume *cdiv1.DataVolume, p
 	if err != nil {
 		return NoClone, err
 	}
+	if bindingMode != nil && *bindingMode == storagev1.VolumeBindingWaitForFirstConsumer {
+		_, isImmediateBindingRequested := datavolume.Annotations[cc.AnnImmediateBinding]
+		waitForFirstConsumerEnabled, err := cc.IsWaitForFirstConsumerEnabled(isImmediateBindingRequested, r.featureGates)
+		if err != nil {
+			return NoClone, err
+		}
+		if !waitForFirstConsumerEnabled {
+			return HostAssistedClone, nil
```

Comment:
The title of this PR is "Respect bind.immediate annotation on fancy clones" but is this not turning a "fancy" clone into a "dumb" clone?  Kind of a misnomer if you ask me.

My expectation was that a dummy pod would consume either:

1.  the temp pvc before namespace transfer
2. the target pvc for same namespace clones

Obviously, there are tradeoffs for either solution, and I'm not suggesting which way is better, but it seems to me that the messaging is off

Reply:
Yep agree the messaging is confusing, will adjust